parser/tcpinfo_test.go

```diff
@@ -169,6 +170,10 @@ func TestTCPParser(t *testing.T) {
 		if row.A.SockID.SrcIP != "195.89.146.242" && row.A.SockID.SrcIP != "2001:5012:100:24::242" {
 			t.Error("Wrong SrcIP", row.A.SockID.SrcIP)
 		}
+		// Guarantee that FinalSnapshot matches the last raw snapshot.
+		if diff := deep.Equal(row.A.FinalSnapshot, row.Raw.Snapshots[len(row.Raw.Snapshots)-1]); diff != nil {
+			t.Errorf("TestTCPParser.ParseAndInsert() FinalSnapshot and last snapshot differ: %s", strings.Join(diff, "\n"))
+		}
```

Comment:
Ah, nice, I think this is a very good way to test this. I had one additional thought:

Maybe we should check that the number of thinned snapshots is correct (per row, not in summary as before)?

If it helps, the expected number of thinned snapshots is: `2 + math.floor((len(row.Raw.Snapshots) - 2) / 10)`, i.e. the first and last element are always included, and every tenth other element.

N.B.: I am not 100% sure how thinSnaps behaves when raw.Snapshots is only 1 element long. Will this element be duplicated? It should not happen, but I am not 100% sure.